mitodl/mit-open
# Add school model, api endpoint; remove school-less department channels

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/3977
Closes https://github.com/mitodl/hq/issues/4031


### Description (What does it do?)
- Adds a `LearningResourceSchool` model with fixtures
- Adds an API endpoint for schools
- Adds `school` to the department serializer and search index schema
- Modifies mgmt commands and plugins to delete channels for departments w/out schools, and not-autocreate such channels in the future.


### How can this be tested?
- On the main branch, if you haven't already done so, run  `./manage.py backpopulate_resource_channels`.  You should end up with 37 department channels (http://localhost:8063/api/v0/channels/?channel_type=department)
- Switch to this branch, restart containers
- Run `./manage.py backpopulate_departments_schools`
- Go to http://localhost:8063/api/v1/schools/.  It should return 6 schools, each with a list of departments under it.  Each department should have `department_id`, `name`, and `channel_url` fields but no `school` field.
- Go to http://localhost:8063/api/v1/departments/.  It should return 37 departments,   Each department should have `department_id`, `name`, `channel_url` and`school` fields.  The `school` field may be null for some, for others it should be a dict containing `id`, `name`, and `url` fields but no `departments` field.
- Run `./manage.py backpopulate_resource_channels`.  It should complete without errors.
- Go to http://localhost:8063/api/v0/channels/?channel_type=department.  There should be 31 results, not 37.
